src/storage-client/src/controller.rs

```diff
@@ -1299,30 +1347,58 @@ where
             if desc.sink.from != *from_id {
                 return Err(StorageError::SourceIdReused(*id));
             }
-            if self
-                .state
-                .exported_collections
-                .get(from_id)
-                // Internal logic error NOT due to export not existing
-                .expect("Dangling exported collection")
-                .iter()
-                .find(|from_export_id| *from_export_id == id)
-                .is_none()
-            {
-                return Err(StorageError::SourceIdReused(*id));
-            }
         }
 
-        for (CreateExportToken { id, from_id }, description) in exports {
-            self.state
-                .exports
-                .insert(id, ExportState::new(description.clone()));
+        for (
+            CreateExportToken {
+                id,
+                from_id,
+                acquired_since,
+            },
+            description,
+        ) in exports
+        {
+            // TODO(aljoscha): I _think_ this description is subtly wrong. But
+            // not completely sure.
+            //
+            // It's worth adding a quick note on write frontiers here.
+            //
+            // The write frontier that sinks communicate back to the controller
+            // indicates that all further writes will happen at a time `t` such
+            // that `!timely::ParitalOrder::less_than(&t, &write_frontier)` is
+            // true.  On restart, the sink will receive an SinkAsOf from this
+            // controller indicating that it should ignore everything at or
+            // before the `since` of the from collection. This will not miss any
+            // records because, if there were records not yet written out that
+            // have an uncompacted time of `since`, the write frontier
+            // previously reported from the sink must be less than `since` so we
+            // would not have compacted up to `since`! This is tested by the
+            // kafka persistence tests.
+            //
+            // Aljoscha: I think we don't need to step back here. The previous
```

Comment:
:skull: Take special care when reading this!

Reply:
Ugh. That sounds right, but I'll need to rummage around a bit to feel more confident.

Thinking about the equivalent situation in compute: a trivial MV with a single input collection. The frontier of that output will advance all the way to the frontier of the input, right? (And the only reason we can't do that for a sink is the "snapshot problem" - that we can't distinguish between updates from immediately before the as-of and updates that happened just after.)

Reply:
Now rather more confident, though I'm still interested in your thought on the MV question!

Reply:
@petrosagg I agree! Sinks are also weird in the frontier they write out. I think they will still need to write out the `F-1` frontier, because that's what they currently do and we can't easily change it now.